File: python/src/company/Company.py
from typing import List
from src.company import Employee, Gender


class Company:
    def __init__(self,company_name: str, founder: Employee) -> None:
        self._company_name = company_name
        self._founder = founder
        self._employee_book = {}
        self._employee_book[founder.get_name()] = founder
        pass

    def get_company_name(self) -> str:
        self._company_name

    def register_employee(self, employee_name: str, gender: Gender) -> Employee:
        pass

    def assign_manager(self,employee_name : str, manager_name: str) -> None:
        pass

    def get_employee(self,employee_name) -> Employee:
        pass

    def get_direct_reports(self,manager_name: str) -> List[Employee]:
        pass

    def get_team_mates(self,employee_name: str) -> List[Employee]:
        pass

    def delete_employee(self,employee_name: str) -> None:
        pass

    def get_employee_hierarchy(self,manager_name: str) -> List[List[Employee]]:
        pass



    




    

    

File: java/src/main/java/com/crio/xcompany/company/Company.java
package com.crio.xcompany.company;

 import java.util.Collections;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;
import java.util.Queue;

public class Company implements ICompany{
    private String companyName;
    private Employee founder;
    private Map<String,Employee> employeeBook;
    

    private Company(String companyName, Employee founder) {
        this.companyName = companyName;
        this.founder = founder;
        employeeBook = new HashMap<String,Employee>();
        employeeBook.put(founder.getName(), founder);
    }
    

    public static Company create(String companyName, Employee founder){
        return new Company(companyName,founder);
    } 


    public String getCompanyName() {
        return companyName;
    }


    @Override
    public Employee registerEmployee(String employeeName, Gender gender) {
     return null;
    }


    @Override
    public void assignManager(String employeeName, String managerName) {
    }


    @Override
    public Employee getEmployee(String employeeName) { 
     return null;
    }


    @Override
    public List<Employee> getDirectReports(String managerName) {
     return Collections.emptyList();
    }


    @Override
    public List<Employee> getTeamMates(String employeeName) {
     return Collections.emptyList();
    }


    @Override
    public void deleteEmployee(String employeeName) {
    }

    @Override
    public List<List<Employee>> getEmployeeHierarchy(String managerName) {
     return Collections.emptyList();
    }
}
